pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,693 @@
+        /// Only auction creator can cancel the auction
+        OnlyAuctionCreatorCanCancel,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// An example dispatchable that takes a singles value as a parameter, writes the value to
+        /// storage and emits an event. This function must be dispatched by a signed extrinsic.
+        #[pallet::weight(T::WeightInfo::create_auction())]
+        #[transactional]
+        pub fn create_auction(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            marketplace_id: MarketplaceId,
+            #[pallet::compact] start_block: T::BlockNumber,
+            #[pallet::compact] end_block: T::BlockNumber,
+            start_price: BalanceOf<T>,
+            buy_it_price: Option<BalanceOf<T>>,
+        ) -> DispatchResultWithPostInfo {
+            let creator = ensure_signed(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+            // ensure the auction timeline is valid
+            ensure!(
+                start_block < end_block,
+                Error::<T>::AuctionCannotStartBeforeItHasEnded
+            );
+            // ensure start block > current block
+            ensure!(
+                start_block > current_block,
+                Error::<T>::AuctionMustStartInFuture
+            );
+            // ensure maximum auction duration is not exceeded
+            ensure!(
+                end_block.checked_sub(&start_block) <= Some(T::MaxAuctionDuration::get()),
+                Error::<T>::AuctionTimelineGreaterThanMaxDuration
+            );
+            // ensure minimum auction duration is valid
+            ensure!(
+                end_block.checked_sub(&start_block) >= Some(T::MinAuctionDuration::get()),
+                Error::<T>::AuctionTimelineLowerThanMinDuration
+            );
+            // ensure buffer period is valid
+            ensure!(
+                start_block.checked_sub(&current_block) >= Some(T::MinAuctionBuffer::get()),
+                Error::<T>::AuctionTimelineBeforeBuffer
+            );
+
+            // ensure the buy_it_price is greater than start_price
+            match buy_it_price {
+                Some(price) => ensure!(price > start_price, Error::<T>::AuctionPricingInvalid),
+                None => (),
+            }
+
+            // fetch the data of given nftId
+            let nft_data = T::NFTHandler::get_nft(nft_id).ok_or(Error::<T>::NFTIdInvalid)?;
+
+            // ensure the caller is the owner of NFT
+            ensure!(nft_data.owner == creator.clone(), Error::<T>::NftNotOwned);
+
+            // ensure the nft is not listed for sale
+            ensure!(
+                nft_data.listed_for_sale == false,
+                Error::<T>::NFTAlreadyListedForSale
+            );
+
+            // ensure the nft is not in transmission
+            ensure!(
```

Comment:
You are missing once more check. You cannot auction nfts that are converted to capsules.